src/filter/elastic_scale/elastic_scale.cpp

```diff
@@ -174,51 +174,57 @@ class ElasticScale : public frei0r::filter
         m_prev_linearScaleFactor = m_linearScaleFactor;
         m_prev_nonLinearScaleFactor = m_nonLinearScaleFactor;
 
-        // transform to internal parameters
+        // map to internal parameters
         m_intern_scaleCenter = m_scaleCenter;
         m_intern_linearScaleArea = m_linearScaleArea;
         m_intern_linearScaleFactor = m_linearScaleFactor;
         m_intern_nonLinearScaleFactor = m_nonLinearScaleFactor*0.4-0.2;
 
         // calculate borders based on parameters
-        m_borderXAbsLeftSrc = int(m_intern_scaleCenter * m_width - m_intern_linearScaleArea / 2 * m_width);
-        m_borderXAbsLeftDst = int(m_intern_scaleCenter * m_width - m_intern_linearScaleArea / 2 * m_width * m_intern_linearScaleFactor);
+        // use 'int' instead of 'unsigned int' as possible results can be negative
+        int tmp_borderXAbsLeftSrc = (int)(m_intern_scaleCenter * width - m_intern_linearScaleArea / 2 * width);
+        int tmp_borderXAbsLeftDst = (int)(m_intern_scaleCenter * width - m_intern_linearScaleArea / 2 * width * m_intern_linearScaleFactor);
 
-        m_borderXAbsRightSrc = int(m_intern_scaleCenter * m_width + m_intern_linearScaleArea / 2 * m_width);
-        m_borderXAbsRightDst = int(m_intern_scaleCenter * m_width + m_intern_linearScaleArea / 2 * m_width * m_intern_linearScaleFactor);
+        int tmp_borderXAbsRightSrc = (int)(m_intern_scaleCenter * width + m_intern_linearScaleArea / 2 * width);
+        int tmp_borderXAbsRightDst = (int)(m_intern_scaleCenter * width + m_intern_linearScaleArea / 2 * width * m_intern_linearScaleFactor);
 
-        // validate if resulting borders are still within the valid range for later calculation [1,m_width-1]
-        if (m_borderXAbsLeftSrc <= 1)
+        // validate if resulting borders are still within the valid range for later calculations [1,width-1]
+        // negative results are limited
+        if (tmp_borderXAbsLeftSrc <= 1)
         { m_borderXAbsLeftSrc = 1; }
-        else if (m_borderXAbsLeftSrc >= m_width-1)
-        { m_borderXAbsLeftSrc = m_width-1; }
+        else if (tmp_borderXAbsLeftSrc >= (int)width-1)
+        { m_borderXAbsLeftSrc = width-1; }
+        else { m_borderXAbsLeftSrc = tmp_borderXAbsLeftSrc; }
```

Comment:
IMO these tmp_ variables are not needed. Wouldn't it make sense to just fix the compare like this:
`if ((int)m_borderXAbsLeftSrc <= 1)`